From 6db8202f7c87ae2cd830acdf920081b824c7b318 Mon Sep 17 00:00:00 2001
From: "Ross-Ashikyan, William" <washikyan@maxlinear.com>
Date: Mon, 4 Nov 2024 11:28:58 -0800
Subject: [PATCH] WLANRTSYS-85310 Fix for 9756cf3276e

Issue: ACS is triggered when hostapd receives a message indicating driver is updated
with correct power mode and regd. Code was originally checking specifically against Standard
Power and not against Indoor Standard Power as well, so ACS was not being triggered
after AFC report was received and parsed under that power mode.

Fix: Generalize if check to work for SP and Indoor SP

FIXUP to WLANRTSYS-82700 mxl changes for V7[4/4] AFC  patches
---
 src/ap/mxl_events.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/ap/mxl_events.c b/src/ap/mxl_events.c
index 094c46c1d..0f75c114f 100644
--- a/src/ap/mxl_events.c
+++ b/src/ap/mxl_events.c
@@ -158,15 +158,16 @@ static int mxl_hostapd_event_6ghz_regdb_update_info(struct hostapd_data *hapd, u
          * AFC_UPDATE_STATUS_DEFAULT. The default status involves marking all 6GHz channels as NO-IR, thereby
          * stopping transmission. Hostap utilizes this information for future decision-making
          * For example, if the AP is in LPI due to fallback, AFC_UPDATE_STATUS_SERVER can trigger a switch to
-         * SP mode; otherwise, it remains in LPI mode.
+         * SP mode; otherwise, it remains in LPI mode. Indoor Standard Power also requires this update
+         * because the driver will trigger this event once it recieves AFC infomation, and ACS needs to run.
          */
 	wpa_printf(MSG_INFO,"%s: Reg Update event received, Configured_mode: %s, Current op_power_mode: %s",
 		   hapd->iface->conf ? hapd->iface->conf->bss[0]->iface : "N/A",
 		   mxl_he_6ghz_pwr_mode_text(hapd->iface->conf->he_6ghz_reg_pwr_type),
 		   mxl_he_6ghz_pwr_mode_text(hapd->iface->mxl_iface.he_6ghz_reg_pwr_mode));
 
-	if (hapd->iface->conf->he_6ghz_reg_pwr_type == HE_REG_INFO_6GHZ_AP_TYPE_SP)
-		mxl_hostapd_interface_update_power_regd(hapd, HE_REG_INFO_6GHZ_AP_TYPE_SP);
+	if (he_reg_is_sp(hapd->iface->conf->he_6ghz_reg_pwr_type))
+		mxl_hostapd_interface_update_power_regd(hapd, hapd->iface->conf->he_6ghz_reg_pwr_type);
 
 	return 0;
 }
-- 
2.43.0

